### PR TITLE
Add cumsum

### PR DESCRIPTION
**Pull request by agoncharenko1992**

### CODE REVIEW

File: onnx2torch/node_converters/cumsum.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+def _arbitrary_dim_shift_and_insert_zero(
+        input_tensor: torch.Tensor,
+        insert_dim: int,
+) -> torch.Tensor:
+
+    t_shape = input_tensor.shape
+    insert_dim = torch.arange(start=0, end=len(t_shape), dtype=torch.int64)[insert_dim]
+    insert_dim_size = t_shape[insert_dim]
+
+    rh_tensor = torch.index_select(
```
Comment (senysenyseny16) [inline onnx2torch/node_converters/cumsum.py:23]: What does `rh` mean?

File: onnx2torch/node_converters/cumsum.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        input=input_tensor,
+        dim=insert_dim,
+        index=torch.arange(0, insert_dim_size-1, dtype=torch.int32, device=input_tensor.device),
```
Comment (senysenyseny16) [inline onnx2torch/node_converters/cumsum.py:26]: named arguments

File: onnx2torch/node_converters/cumsum.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    )
+
+    insert_index = torch.arange(1, insert_dim_size, dtype=torch.int64, device=input_tensor.device)
```
Comment (senysenyseny16) [inline onnx2torch/node_converters/cumsum.py:29]: named arguments

File: onnx2torch/node_converters/cumsum.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    insert_index = insert_index + torch.zeros_like(rh_tensor, dtype=torch.int64, device=input_tensor.device)
+
+    input_tensor = torch.scatter(
+        input=input_tensor,
+        dim=insert_dim,
+        index=insert_index,
+        src=rh_tensor,
+    )
+
+    shuffled_input_tensor = torch.transpose(input_tensor, len(t_shape) - 1, insert_dim)
```
Comment (senysenyseny16) [inline onnx2torch/node_converters/cumsum.py:43]: named arguments

File: onnx2torch/node_converters/cumsum.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    shuffled_input_tensor[..., 0] = 0
+    input_tensor = torch.transpose(shuffled_input_tensor, len(t_shape) - 1, insert_dim)
```
Comment (senysenyseny16) [inline onnx2torch/node_converters/cumsum.py:45]: named arguments

File: onnx2torch/node_converters/cumsum.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+from onnx2torch.utils.common import OnnxToTorchModule
+from onnx2torch.utils.common import OperationConverterResult
+from onnx2torch.utils.common import onnx_mapping_from_node
+
+
+def _arbitrary_dim_shift_and_insert_zero(
+        input_tensor: torch.Tensor,
+        insert_dim: int,
+) -> torch.Tensor:
+
```
Comment (senysenyseny16) [inline onnx2torch/node_converters/cumsum.py:18]: rm blank line

File: onnx2torch/node_converters/cumsum.py (hunk 3 above)
Comment (agoncharenko1992) [reply onnx2torch/node_converters/cumsum.py]: done.

File: onnx2torch/node_converters/cumsum.py (hunk 4 above)
Comment (agoncharenko1992) [reply onnx2torch/node_converters/cumsum.py]: done.

File: onnx2torch/node_converters/cumsum.py (hunk 5 above)
Comment (agoncharenko1992) [reply onnx2torch/node_converters/cumsum.py]: done.

File: onnx2torch/node_converters/cumsum.py (hunk 6 above)
Comment (agoncharenko1992) [reply onnx2torch/node_converters/cumsum.py]: done.

File: onnx2torch/node_converters/cumsum.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+from onnx2torch.utils.common import OperationConverterResult
+from onnx2torch.utils.common import onnx_mapping_from_node
+
+
+def _arbitrary_dim_shift_and_insert_zero(
+        input_tensor: torch.Tensor,
+        insert_dim: int,
+) -> torch.Tensor:
+    t_shape = input_tensor.shape
+    insert_dim = torch.arange(start=0, end=len(t_shape), dtype=torch.int64)[insert_dim]
```
Comment (irkjero) [inline onnx2torch/node_converters/cumsum.py:19]: Why do we need only positive insert_dim. Something of index_select, scatter or transpose doesnt work with negative dims?

Comment (agoncharenko1992) [reply onnx2torch/node_converters/cumsum.py:19]: done,

**Review by irkjero — APPROVED** [review]

**Pull request merged**